src/utils/accountService.js

```diff
@@ -0,0 +1,387 @@
+   * Async creation results in faster account creation regardless of the AccountID -> privateKey order.
+   * @param {boolean} async
+   * @param {number} balance
+   * @param {number} num
+   * @param {boolean} startup
+   */
+  async generate(async, balance, num, startup) {
+    if (async) {
+      this.logger.log("Generating accounts in asynchronous mode...");
+      await this.generateAsync(balance, num, startup);
+    } else {
+      this.logger.log("Generating accounts in synchronous mode...");
+      await this.generateSync(balance, num, startup);
+    }
+  }
+
+  /**
+   * Generate accounts synchronous, with correct accountID -> privateKey mapping.
+   * @param {number} balance
+   * @param {number} num
+   * @param {boolean} startup
+   */
+  async generateSync(balance, num, startup) {
+    await this._generateECDSA(false, balance, num, startup);
+    this.logger.log("");
+    await this._generateAliasECDSA(false, balance, num, startup);
+    this.logger.log("");
+    await this._generateED25519(false, balance, num, startup);
+  }
+
+  /**
+   * Generate accounts asynchronous.
+   * @param {number} balance
+   * @param {number} num
+   * @param {boolean} startup
+   */
+  async generateAsync(balance, num, startup = false) {
+    Promise.all([
+      await this._generateECDSA(true, balance, num, startup),
+      await this._generateAliasECDSA(true, balance, num, startup),
+      await this._generateED25519(true, balance, num, startup)
+    ]).then((allResponses) => {
+      const ecdsaResponses = allResponses[0];
+      const aliasEcdsaResponses = allResponses[1];
+      const ed25519Responses = allResponses[2];
+
+      this._logAccountTitle();
+      ecdsaResponses.forEach(element => {
+        this._logAccount(element.accountNum, element.balance, element.wallet._signingKey().privateKey);
+      });
+      this._logAccountivider();
+
+      this._logAliasAccountTitle();
+      aliasEcdsaResponses.forEach(element => {
+        this._logAliasAccount(element.accountNum, element.balance, element.wallet);
+      });
+      this._logAliasAccountDivider();
+
+      this._logAccountTitle();
+      ed25519Responses.forEach(element => {
+        this._logAccount(element.accountNum, element.balance, element.wallet._signingKey().privateKey);
+      });
+      this._logAccountivider();
+    })
+  }
+
+  /**
+   * @internal
+   * Generate ECDSA accounts.
+   * @param {boolean} async
+   * @param {number} balance
+   * @param {number} num
+   * @param {boolean} startup
+   */
+  async _generateECDSA(async, balance, num, startup) {
+    let ecdsaAccountNumCounter = 1002;
+    const accounts = [];
+
+    if (!async) this._logAccountTitle();
+
+    for (let i = 0; i < num; i++) {
+      let wallet = hethers.Wallet.createRandom();
+      if (startup && this.privateKeysECDSA[i]) {
+        wallet = new hethers.Wallet(this.privateKeysECDSA[i]);
+      }
+      if (async) {
+        accounts.push(() => this._createAccount(
```

Comment:
I've updated this, removed the anonymous func. Also updated the desc, almost no difference, but this is because of the recent update to sync creation of accounts. Not sure if you saw, but is considerably faster now.

Reply:
@georgi-l95 as I see, I misunderstood what you meant by:

> This new mode makes startup faster by around 3 seconds.

as it only decreased the startup time by about 3s, which was not as much as I expected. But I saw you edits, so it's effectively 14s faster 🚀